### PR TITLE
[SELS-TASK] Implemented Add Word Function for Admin

### PR DESCRIPTION
**Pull request by christanplaza**
Description: This PR is a Back-End implementation for Add Word Function for the Admin Panel.

[Commands]
- php artisan migrate:fresh --seed

[Pre-condition]
- Login as Admin (Username: admin@gmail.com, Password: securepassword)
- Proceed to /admin/categories/
- Add a Word by clicking on 'Add Word'
- Input word text
- Input the four choices
- Click the radiobutton for the Correct answer
- Click Save
- Click the category that you added a word to

[Expected Output]
- The page should redirect to /admin/categories
- After going to the Category Page, the new word should be in the list

[Notes]
- The redirect is fixed in a different PR where the page would redirect to the category that was added a word to.

### CODE REVIEW

File: app/Http/Controllers/AdminWordsController.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public function create(Category $category)
+    {
+        return view('admin.word.create', compact('category'));
+    }
+
+    public function store(Category $category){
+        $attributes = $this->validateEntry();
+        $attributes['category_id'] = $category->id;
+        $word = new Word;
+        $word->newRecord($attributes);
```
Comment (jeremiahC) [inline app/Http/Controllers/AdminWordsController.php]: one trick to make this oneline is that you can do this `(new Word)->newRecord($attributes);`

File: app/Http/Controllers/AdminWordsController.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+        return redirect('/admin/categories');
+    }
+
+    protected function validateEntry(){
```
Comment (jeremiahC) [inline app/Http/Controllers/AdminWordsController.php]: You can create a Form request file for your validation(https://laravel.com/docs/5.8/validation) and just call as parameter in `store()`.

Comment (jeremiahC) [reply app/Http/Controllers/AdminWordsController.php]: It will automatically validate

File: app/Http/Controllers/AdminWordsController.php (hunk 1 above)
Comment (christanplaza) [reply app/Http/Controllers/AdminWordsController.php]: To my understanding, line 19 should be `$word = (new Word)->newRecord($attributes);` ?

File: app/Http/Controllers/AdminWordsController.php (hunk 2 above)
Comment (christanplaza) [reply app/Http/Controllers/AdminWordsController.php]: I will work on this immediately. Please allow me to finish my current PR.

File: app/Http/Controllers/AdminWordsController.php (hunk 1 above)
Comment (jeremiahC) [reply app/Http/Controllers/AdminWordsController.php]: yes

File: routes/web.php [hunk 3]
```diff
@@ -21,4 +21,7 @@
 
 Route::resource('/admin/categories', 'AdminCategoryController');
 
+Route::post('/admin/categories/{category}/words', 'AdminWordsController@store');
```
Comment (jeremiahC) [inline routes/web.php]: Use `group` function or `prefix` function in routes to group related routes

File: app/Word.php [hunk 4]
```diff
@@ -16,4 +18,14 @@ public function choices()
...
     }
+
+    public function newRecord($attributes){
+        $record = $this->create([
+            'category_id' => $attributes['category_id'],
+            'text' => $attributes['text'],
+        ]);
+        $attributes['word_id'] = $record->id;
+        $choices = new Choice;
+        $choices->newRecord($attributes);
```
Comment (jeremiahC) [inline app/Word.php]: You are passing the attributes array, but there are unnecessary part which is `category_id` and `text`, it is not related to choices anymore. You can use laravel collection to filter the array and just pass what is important.

File: app/Http/Controllers/AdminWordsController.php (hunk 1 above)
Comment (christanplaza) [reply app/Http/Controllers/AdminWordsController.php]: would `(new Word)->newRecord($attributes);` be alright as well?

**Comment by jeremiahC** [comment]
@christanplaza 
Can you add validation message, although you have validation but Admin cannot know if there is required or something error.

**Comment by christanplaza** [comment]
> @christanplaza
> Can you add validation message, although you have validation but Admin cannot know if there is required or something error.

Will do. Give me a minute.

**Comment by christanplaza** [comment]
> @christanplaza
> Can you add validation message, although you have validation but Admin cannot know if there is required or something error.

This has been resolved in 749873b

File: app/Http/Controllers/AdminWordsController.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+use App\Http\Requests\CreateWord;
+
+class AdminWordsController extends Controller
+{
+    public function create(Category $category)
+    {
+        return view('admin.word.create', compact('category'));
+    }
+
+    public function store(Category $category, CreateWord $request){
```
Comment (jeremiahC) [inline app/Http/Controllers/AdminWordsController.php]: open brace, must be next line

**Comment by jeremiahC** [comment]
@christanplaza 
Add success message

File: app/Http/Controllers/AdminWordsController.php (hunk 5 above)
Comment (christanplaza) [reply app/Http/Controllers/AdminWordsController.php]: Resolved in eef3a03

**Comment by christanplaza** [comment]
> @christanplaza
> Add success message

eef3a03 Resolved

**Review by jeremiahC — APPROVED** [review]

**Pull request merged**